orangecontrib/educational/widgets/owkmeans.py

```diff
@@ -0,0 +1,532 @@
+                                               maxv=10,
+                                               step=1,
+                                               label='Number of centroids:',
+                                               callback=self.number_of_clusters_change)
+        self.centroidNumbersSpinner.setSizePolicy(QSizePolicy(QSizePolicy.MinimumExpanding, QSizePolicy.Fixed))
+        self.restartButton = gui.button(self.centroidsBox, self, self.button_labels["random_centroids"],
+                                        callback=self.restart)
+        self.linesCheckbox = gui.checkBox(self.centroidsBox,
+                                          self,
+                                          'lines_to_centroids',
+                                          'Show membership lines',
+                                          callback=self.complete_replot)
+
+        # control box
+        self.commandsBox = gui.widgetBox(self.controlArea)
+        self.stepButton = gui.button(self.commandsBox, self, self.button_labels["step2"],
+                                     callback=self.step)
+        self.stepBackButton = gui.button(self.commandsBox, self, self.button_labels["step_back"],
+                                         callback=self.step_back)
+        self.autoPlayButton = gui.button(self.commandsBox, self, self.button_labels["autoplay_run"],
+                                         callback=self.auto_play)
+        self.autoPlaySpeedSpinner = gui.hSlider(self.commandsBox,
+                                                self,
+                                                'autoPlaySpeed',
+                                                minValue=0,
+                                                maxValue=1.91,
+                                                step=0.1,
+                                                intOnly=False,
+                                                createLabel=False,
+                                                label='Speed:')
+
+        gui.rubber(self.controlArea)
+
+        # disable until data loaded
+        self.set_disabled_all(True)
+
+        # graph in mainArea
+        self.scatter = Scatterplot(click_callback=self.graph_clicked,
+                                   drop_callback=self.centroid_dropped,
+                                   xAxis_gridLineWidth=0,
+                                   yAxis_gridLineWidth=0,
+                                   title_text='',
+                                   tooltip_shared=False,
+                                   debug=True)  # TODO: set false when end of development
+        # Just render an empty chart so it shows a nice 'No data to display'
+        self.scatter.chart()
+        self.mainArea.layout().addWidget(self.scatter)
+
+    def concat_x_y(self):
+        """
+        Function takes two selected columns from data table and merge them in new Orange.data.Table
+        :return: table with selected columns
```

Comment:
We tend to prefer numpy-style napoleon docstrings. You can tell PyCharm to use different style in Python Integrated Tools section of the Preferences.